dataservice/api/read_group/models.py

```diff
@@ -43,20 +48,20 @@ class ReadGroupGenomicFile(db.Model, Base):
     __tablename__ = 'read_group_genomic_file'
     __prefix__ = 'RF'
     __table_args__ = (db.UniqueConstraint('read_group_id',
-                                          'genomic_file_id'),)
+                                          'genomic_file_id',),)
     read_group_id = db.Column(KfId(),
                               db.ForeignKey('read_group.kf_id'),
                               nullable=False)
 
     genomic_file_id = db.Column(KfId(),
                                 db.ForeignKey('genomic_file.kf_id'),
                                 nullable=False)
-    read_group = db.relationship('ReadGroup')
-    genomic_file = db.relationship('GenomicFile')
+    external_id = db.Column(db.Text(),
+                            doc='external id used by contributor')
 
 
-@event.listens_for(GenomicFile, 'after_delete')
+@event.listens_for(ReadGroupGenomicFile, 'after_delete')
```

Comment:
Is this still necessary? I would think that proper cascading would deal with this appropriately.

Reply:
@dankolbman Yea we still need this. Remember cascade delete orphans are usually for child concepts like ReadGroupGenomicFile, not parent concepts like ReadGroup. So we need something that will listen for when all children of the parent are deleted, and then delete the parent since it will be an orphan at that point.